### ISSUE TITLE
Failing user hooks report wrong number of errors in Igor Pro

### ISSUE THREAD

**Issue by t-b** [byte-physics/igor-unit-testing-framework#55]
Running `AbortingUserHooks.pxp` gives

```
•run()
  Start of test "Unnamed"
  Encountered "Abort" in user hook "TEST_BEGIN_OVERRIDE", procedure file "Undefined Procedure"
  Entering test suite "aborting-user-hooks.ipf"
  Encountered "Abort" in user hook "TEST_SUITE_BEGIN_OVERRIDE", procedure file "aborting-user-hooks.ipf"
  Entering test case "Dostuff"
  Encountered "Abort" in user hook "TEST_CASE_BEGIN_OVERRIDE", procedure file "aborting-user-hooks.ipf"
  Encountered "Abort" in user hook "TEST_CASE_END_OVERRIDE", procedure file "aborting-user-hooks.ipf"
  Leaving test case "TEST_CASE_END_OVERRIDE"
  Encountered "Abort" in user hook "TEST_SUITE_END_OVERRIDE", procedure file "aborting-user-hooks.ipf"
  Failed with 4 errors
  Leaving test suite "TEST_SUITE_END_OVERRIDE"
  Encountered "Abort" in user hook "TEST_END_OVERRIDE", procedure file "Undefined Procedure"
  Test finished with 4 errors
  End of test "TEST_END_OVERRIDE"
```

but I would have expected 6. The error is somewhere in the error_count clearing in `TestSuiteBegin`.

**Comment by Garados007** [comment]
If I run this with PR #381 (177ec83 (Treat user hooks as test cases for the results, 2023-01-13)) I get the following results:

```
•run()
  Start of test "Unnamed"
  Encountered "Abort" in user hook "TEST_BEGIN_OVERRIDE" (No source procedure)
  Finished with no errors
  Leaving test suite "aborting-user-hooks.ipf"
  Encountered "Abort" in user hook "TEST_END_OVERRIDE" (No source procedure)
  Test finished with 2 errors
    ▶ Encountered "Abort" in user hook "TEST_BEGIN_OVERRIDE" (No source procedure)
    ▶ Encountered "Abort" in user hook "TEST_END_OVERRIDE" (No source procedure)
  End of test "Unnamed"
```

It seems like that an abort in a user hook aborts the whole test run. Is this intended or should it be fixed?

**Comment by Garados007** [comment]
The behavior change was introduced in 4f274fb (Adapted main loop to index over test run data wave instead of two loops, 2022-09-14).

**Comment by Garados007** [comment]
According to our latest discussion this behavioral change was intended and the error counter was corrected with 0a18141 (Treat user hooks as test cases for the results, 2023-01-13). In conclusion the topic of this issue is solved and can be closed.
